ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/AttestationUtil.java

```diff
@@ -46,14 +48,20 @@
         epoch_attestations.add(attestation);
       }
     }
-    if (epoch_attestations.isEmpty()) {
-      throw new Exception("No pending attestation found for the specified epoch");
-    }
+
+    checkArgument(epoch_attestations.size() == 0, "There are no epoch_attestations");
     return epoch_attestations;
   }
 
+  /**
+   * Returns the current epoch boundary attestations.
+   *
+   * @param state
+   * @return List<PendingAttestation>
+   * @throws IllegalArgumentException
+   */
   public static List<PendingAttestation> get_current_epoch_boundary_attestations(BeaconState state)
-      throws Exception {
+      throws IllegalArgumentException {
```

Comment:
Question out of curiosity: when you specify the Exception type like that, do you get better (more specified) exception print outs?

Reply:
I think it's better to be specific because just throwing Exception essentially boxes the exception type and forces everyone to handle Exception rather than the specific type. Also, you lose the granularity of what type of exception was really thrown

Reply:
Thanks for the explanation.